Generic file processor - attempt 3

robintw opened the pull request:
In progress

IanMayo committed:
remove duplicate layer ids

IanMayo committed:
introduce example with large gap between anchor and content, due to presence of intermediary image

IanMayo committed:
introduce example of image containing space character

IanMayo committed:
replace duplicate image

IanMayo committed:
introduce instance of file with filename that isn't a valid XML id

IanMayo committed:
introduce renamed file

IanMayo committed:
introduce use of `em` element, replace with `i`

IanMayo committed:
Sometimes the `PageLayer` is missing an id.  So, also strip out absolute divs that don't have an id

IanMayo committed:
Switch heading to H2, so it gets caught in our "title finder"

IanMayo committed:
Introduce example of link target within text

IanMayo committed:
move anchor out of heading (target data similarly fixed)

IanMayo committed:
introduce comment

IanMayo committed:
Put the floating anchor into the parent GrayLayer

robintw committed:
Update copy_files to copy recursively (controlled by parameter), so we can sanitise all filenames when copying, removing issues with spaces in image filenames

robintw committed:
Fix error when filenames like `XXXX(12).html` are used, by removing ( and ) when sanitising

robintw committed:
Adds ./ to the start of the URL for links to files in the same directory, so Oxygen doesn't complain. Fixes #439

robintw committed:
Change levels of a few logging messages, plus print out current logging level. Fixes #436

robintw committed:
Add another fallback method for finding the title - looking for a h1 tag. Fixes #431

IanMayo commented:
Hello Robin, I was a bit nervous of changing the filenames for images, for fear of broken links. I "did" only aim to fix the path for DITA files that we generate, but I realise they still receive incoming links.
So yes, let's fix filenames for files we generate as well as files we copy over, and rely on the DITA publish process to spot any holes in our logic.

robintw committed:
Sanitise all filenames, including the DITA files we generate, and fix all links.

As suggested by comment at https://github.com/DeepBlueCLtd/LegacyMan/pull/440#issuecomment-1752026112

Includes changing sanitisation to remove ( and ) rather than replace with _, it makes for tidier filenames

robintw commented:
Done in fe0fd97

robintw committed:
Allow us to find BottomLayers that are inside GrayLayers

Code is a bit messy and repeated at the moment, but I want to check it works on-site before tidying up

IanMayo commented on parser/parser.py:
```diff
@@ -553,15 +553,34 @@ def process_generic_file_content(self, html_soup, input_file_path, quicklinks):
                             if top_value > enclosing_div_top_value:
                                 # Check that the difference isn't too big
-                                if top_value - enclosing_div_top_value < 500:
+                                if top_value - enclosing_div_top_value < 1000:
```
I'm a little nervous about going this large - since we're approaching the vertical separation between separate pages.  Could we try with 800 first please?

robintw replied:
Sure, just tested with 800 and it works. Will commit it now.

robintw committed:
Change the dita executable that is used when on Windows

robintw committed:
Change maximum distance to look for a BottomLayer to 800 from 1000
